src/CowSdk.ts

```diff
@@ -1,13 +1,44 @@
+import log from 'loglevel'
 import { version as SDK_VERSION } from '../package.json'
+import { CowApi } from './api'
+import { SupportedChainId as ChainId } from './constants/chains'
 import { validateAppDataDocument } from './utils/appData'
-import { signOrder, signOrderCancellation } from './utils/sign'
+import { Context, CowContext } from './utils/context'
+import { signOrder, signOrderCancellation, UnsignedOrder } from './utils/sign'
 
-export class CowSdk {
+export class CowSdk<T extends ChainId> {
   static version = SDK_VERSION
+  chainId: T
+  context: Context
+  cowApi: CowApi<T>
+
+  constructor(chainId: T, cowContext: CowContext) {
+    this.chainId = chainId
+    this.context = new Context(cowContext)
+    this.cowApi = new CowApi(chainId, this.context)
+  }
 
   validateAppDataDocument = validateAppDataDocument
-  signOrder = signOrder
-  signOrderCancellation = signOrderCancellation
+
+  signOrder(order: UnsignedOrder) {
+    try {
+      const signer = this.context.provider.getSigner()
+
+      return signOrder(order, this.chainId, signer)
+    } catch (error) {
+      log.error('There was an error signing the order', error)
```

Comment:
I think that, as a library, we should let the error handling up to the consumer.
I would not catch and log error in this case, and instead let it bubble up.